src/storage.rs

```diff
@@ -11,11 +11,12 @@ pub struct Storages<'a> {
 }
 
 impl<'a> Storages<'a> {
-    /// Create storage using default directory
-    pub fn new() -> Storages<'a> {
-        Storages {
-            //TODO use user home subdir
-            base_dir: Path::new("emerald_data"),
+    /// Create storage using user directory if specified,
+    /// or default path in other case.
+    pub fn new(path: Option<&'a Path>) -> Storages<'a> {
+        match path {
+            Some(p) => Storages { base_dir: p },
+            _ => Storages { base_dir: Path::new("emerald_data") },
```

Comment:
Why `emerald_data`, in issue stay:

Mac OS X: ~/Library/Emerald
Linux: ~/.emerald
Windows: %USERDIR%\.emerald